test/fshtypes/Profile.test.ts

```diff
@@ -13,6 +19,27 @@ describe('Profile', () => {
     });
   });
 
+  describe('#id', () => {
+    it('should return an id set by a string caret rule', () => {
+      const p = new Profile('MyObservation');
+      const idRule = new CaretValueRule('');
+      idRule.caretPath = 'id';
+      idRule.value = 'different-id';
+      p.rules.push(idRule);
+      expect(p.id).toBe('different-id');
+    });
+
+    it('should not return an id set by an instance caret rule', () => {
+      const p = new Profile('MyObservation');
+      const idRule = new CaretValueRule('');
+      idRule.caretPath = 'id';
+      idRule.value = 'different-id';
+      idRule.isInstance = true;
+      p.rules.push(idRule);
+      expect(p.id).toBe('MyObservation');
+    });
```

Comment:
Here too. Can we assert no errors in the first test and the expected error in the second test?